histogram.go

```diff
@@ -368,7 +369,7 @@ func ExponentialDurationBuckets(start time.Duration, factor float64, n int) (Dur
 	curr := start
 	for i := range buckets {
 		buckets[i] = curr
-		curr = time.Duration(float64(curr) * factor)
+		curr = time.Duration(safeFloat64ToInt64(float64(curr) * factor))
```

Comment:
options we talked about offline, consider the existing signature 
```
func ExponentialDurationBuckets(start time.Duration, factor float64, n int) (DurationBuckets, error) {
```

we could change this to
```
func ExponentialDurationBuckets(
  start time.Duration, 
  factor float64, 
  n int,
  opts ...BucketOpts,
) (DurationBuckets, error) {
...
}

// you can have opts for "autofix buckets" or "error on invalid bucket range/overflow/etc
```